e2e/cypress/integration/enterprise/system_console/compliance/data_retention_policies_spec.js

```diff
@@ -0,0 +1,449 @@
+            });
+        });
+
+        it('MM-T4006 - Policies count', () => {
+            cy.uiClickCreatePolicy();
+
+            cy.uiFillOutCustomPolicyFields('Policy 1', 'days', '60');
+
+            cy.uiAddRandomChannelToCustomPolicy();
+
+            cy.uiGetButton('Save').click();
+
+            cy.uiClickCreatePolicy();
+
+            cy.uiFillOutCustomPolicyFields('Policy 2', 'days', '160');
+
+            cy.uiAddRandomChannelToCustomPolicy();
+
+            cy.uiGetButton('Save').click();
+
+            cy.uiClickCreatePolicy();
+
+            cy.uiFillOutCustomPolicyFields('Policy 3', 'days', '100');
+
+            cy.uiAddRandomChannelToCustomPolicy();
+
+            cy.uiGetButton('Save').click();
+
+            cy.get('#custom_policy_table .DataGrid .DataGrid_footer .DataGrid_cell').should('be.visible').invoke('text').should('include', '1 - 3 of 3');
+
+            cy.apiGetCustomRetentionPolicies().then((result) => {
+                expect(result.body.total_count).to.equal(3);
+            });
+        });
+
+        it('MM-T4008 - Update custom policy', () => {
+            cy.uiClickCreatePolicy();
+
+            cy.uiFillOutCustomPolicyFields('Policy 2', 'years', '2');
+
+            cy.uiAddRandomTeamToCustomPolicy();
+
+            cy.uiGetButton('Save').click();
+
+            cy.get('#custom_policy_table .DataGrid').should('be.visible');
+
+            cy.wait('@createCustomPolicy').then((interception) => {
+                cy.uiVerifyPolicyResponse(interception.response.body, 1, 0, 730, 'Policy 2');
+                const policyId = interception.response.body.id;
+
+                cy.get('#custom_policy_table .DataGrid').within(() => {
+                    cy.uiVerifyCustomPolicyRow(policyId, 'Policy 2', '2 years', '1 team, 0 channels');
+
+                    cy.uiClickEditCustomPolicyRow(policyId);
+                });
+                cy.get('.DataRetentionSettings .admin-console__header', {timeout: TIMEOUTS.TWO_MIN}).should('be.visible').invoke('text').should('include', 'Custom Retention Policy');
+
+                cy.get('.PolicyTeamsList .DataGrid').within(() => {
+                    cy.findByRole('link', {name: 'Remove'}).should('be.visible').click();
+                });
+
+                cy.uiAddRandomChannelToCustomPolicy();
+
+                cy.uiGetButton('Save').click();
+
+                cy.get('#custom_policy_table .DataGrid').should('be.visible');
+
+                cy.get('#custom_policy_table .DataGrid').within(() => {
+                    cy.uiVerifyCustomPolicyRow(policyId, 'Policy 2', '2 years', '0 teams, 1 channel');
+                });
+
+                cy.apiGetCustomRetentionPolicy(policyId).then((result) => {
+                    expect(result.body.team_count).to.equal(0);
+                    expect(result.body.channel_count).to.equal(1);
+                    expect(result.body.post_duration).to.equal(730);
+                    expect(result.body.display_name).to.equal('Policy 2');
```

Comment:
Don't forget assertion comments and stuff
```
// - [#] indicates a test step (e.g. # Go to a page)
// - [*] indicates an assertion (e.g. * Check the title)
// - Use element ID when selecting an element. Create one if none.
```

Reply:
I added a bunch of comments, let me know if they are ok